Open Judge System/Web/OJS.Web/Areas/Administration/Controllers/ContestsController.cs

```diff
@@ -91,6 +96,19 @@ public ActionResult Create(ContestAdministrationViewModel model)
                 return this.View(model);
             }
 
+            if (model.IsOnline)
+            {
+                if (!model.Duration.HasValue)
+                {
+                    this.ModelState.AddModelError(nameof(model.Duration), Resource.Required_field_for_online);
+                }
+
+                if (model.NumberOfProblemGroups <= 0)
+                {
+                    this.ModelState.AddModelError(nameof(model.NumberOfProblemGroups), Resource.Required_field_for_online);
+                }
+            }
+
```

Comment:
Add additional validation for the `Duration` field as it can easily overflow.

Reply:
Done